agent/containers_dockercli.go

```diff
@@ -0,0 +1,188 @@
+package agent
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coder/coder/v2/codersdk"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/xerrors"
+)
+
+// dockerCLIContainerLister is a ContainerLister that lists containers using the docker CLI
+type dockerCLIContainerLister struct{}
+
+var _ ContainerLister = &dockerCLIContainerLister{}
+
+func (*dockerCLIContainerLister) List(ctx context.Context) ([]codersdk.WorkspaceAgentContainer, error) {
+	var buf bytes.Buffer
+	// List all container IDs, one per line, with no truncation
+	cmd := exec.CommandContext(ctx, "docker", "ps", "--all", "--quiet", "--no-trunc")
+	cmd.Stdout = &buf
+	if err := cmd.Run(); err != nil {
+		return nil, xerrors.Errorf("run docker ps: %w", err)
```

Comment:
We're leaving stderr on the floor here (I think); would it not be useful to send that back, or is that already included in `err`?

Reply:
Yeah, I'll add `stderr` to the error here:

```
run docker ps: exit status 1: "permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get \"http://%2Fvar%2Frun%2Fdocker.sock/v1.47/containers/json?all=1\": dial unix /var/run/docker.sock: connect: permission denied"
```